### PR TITLE
Support cloud sql private ip (incorporating previous PR feedback)

### PR DESCRIPTION
**Pull request by craigatgoogle**
This PR picks up on the work done by @yhuang incorporating the feedback given by @danawillow.

### CODE REVIEW

**Comment by craigatgoogle** [comment]
For context, a link to the previous PR: https://github.com/terraform-providers/terraform-provider-google-beta/pull/46

File: google-beta/resource_sql_database_instance_test.go [hunk 1]
```diff
@@ -594,6 +594,30 @@ func TestAccSqlDatabaseInstance_basic_with_user_labels(t *testing.T) {
 	})
 }
 
+func TestAccSqlDatabaseInstance_with_private_network(t *testing.T) {
```
Comment (danawillow) [inline google-beta/resource_sql_database_instance_test.go]: nit: could it be called `TestAccSqlDatabaseInstance_withPrivateNetwork`?

Comment (craigatgoogle) [reply google-beta/resource_sql_database_instance_test.go]: Done.

File: google-beta/resource_sql_database_instance_test.go [hunk 2]
```diff
@@ -714,6 +738,42 @@ resource "google_sql_database_instance" "instance-failover" {
 `, instanceName, failoverName)
 }
 
+func testAccSqlDatabaseInstance_with_private_network(databaseName, networkName, addressRangeName string) string {
+	return fmt.Sprintf(`
+resource "google_compute_network" "foobar" {
```
[the remark this answers is not in the document]
Comment (craigatgoogle) [reply google-beta/resource_sql_database_instance_test.go]: Done.

File: google-beta/resource_sql_database_instance_test.go [hunk 3]
```diff
@@ -714,6 +738,42 @@ resource "google_sql_database_instance" "instance-failover" {
...
+	network       = "${google_compute_network.foobar.self_link}"
+}
+
+resource "google_service_networking_connection" "foobar" {
+	network       = "${google_compute_network.foobar.self_link}"
+	service       = "servicenetworking.googleapis.com"
+	reserved_peering_ranges = ["${google_compute_global_address.foobar.name}"]
+}
+
+# TODO figure out a way to specify the dependency to the connection resource
```
[the remark this answers is not in the document]
Comment (craigatgoogle) [reply google-beta/resource_sql_database_instance_test.go]: Removed.

**Comment by craigatgoogle** [comment]
Thanks for taking a look :)

**Comment by craigatgoogle** [comment]
Documentation added.

**Comment by craigatgoogle** [comment]
Closing this out. Thanks again for the help.

**Pull request merged**